src/java.base/macosx/classes/sun/nio/fs/BsdFileSystemProvider.java

```diff
@@ -58,6 +58,16 @@ public <V extends FileAttributeView> V getFileAttributeView(Path obj,
             return (V) new BsdUserDefinedFileAttributeView(UnixPath.toUnixPath(obj),
                     Util.followLinks(options));
         }
+        if (type == BasicFileAttributeView.class) {
+            UnixPath file = UnixPath.toUnixPath(obj);
+            boolean followLinks = Util.followLinks(options);
+            return (V) BsdFileAttributeViews.createBasicView(file, followLinks);
+        }
+        if (type == PosixFileAttributeView.class) {
+            UnixPath file = UnixPath.toUnixPath(obj);
+            boolean followLinks = Util.followLinks(options);
+            return (V) BsdFileAttributeViews.createPosixView(file, followLinks);
+        }
```

Comment:
The update that is 10deb03d looks quite good. I just wonder about the duplicate in BsdFileSystemProvider. Maybe it could test for UserDefiendFileAttributeFIleView, BasicFileAttrinuteView or PosixFileAttributeView so that the parameters are only processed in one place rather than 3.

Reply:
Agreed, I don't like that either.

Reply:
Revised in 0274bc5523310e0b8580be4f63acec51f0436c49.